android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/dashboard/DashboardFragment.java

```diff
@@ -1,80 +1,239 @@
+import java.time.temporal.ChronoField;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import de.dhbw.ka.se.fibo.ApplicationState;
 import de.dhbw.ka.se.fibo.R;
 import de.dhbw.ka.se.fibo.databinding.FragmentDashboardBinding;
-import java.util.Date;
-
-public class DashboardFragment extends Fragment {
-
-  private FragmentDashboardBinding binding;
-  private MaterialDatePicker<Pair<Long, Long>> picker;
-  private Date startDate;
-  private Date endDate;
-
-
-  public View onCreateView(@NonNull LayoutInflater inflater,
-      ViewGroup container, Bundle savedInstanceState) {
-    DashboardViewModel dashboardViewModel =
-        new ViewModelProvider(this).get(DashboardViewModel.class);
-
-    binding = FragmentDashboardBinding.inflate(inflater, container, false);
-    View root = binding.getRoot();
-    initializeDateCard();
-    createDatePicker();
-    return root;
-  }
-
-  private void initializeDateCard() {
-    setDateCardTitle();
-    setDateCardTime();
-    setListener();
-  }
-
-  private void setListener() {
-    binding.button.setOnClickListener(
-        e -> picker.show(requireActivity().getSupportFragmentManager(), "date_pange_picker"));
-  }
-
-  private void setDateCardTime() {
-    if (null == startDate) {
-      startDate = new Date();
-      startDate.setMonth(startDate.getMonth() - 1);
+import de.dhbw.ka.se.fibo.models.Cashflow;
+import de.dhbw.ka.se.fibo.models.Category;
+
+public class DashboardFragment extends Fragment implements OnChartValueSelectedListener {
+
+    private FragmentDashboardBinding binding;
+    private MaterialDatePicker<Pair<Long, Long>> picker;
+    private LocalDate startDate = null;
+    private LocalDate endDate = null;
+    private PieChart pieChart;
+
+    private static final int PIE_CHART_ANIMATION_DURATION = 750;
+
+    public View onCreateView(@NonNull LayoutInflater inflater,
+                             ViewGroup container, Bundle savedInstanceState) {
+        // gather current instant to be able to calculate the time we spent in this (user-interaction driven) method
+        Instant instant = Instant.now();
+
+        DashboardViewModel dashboardViewModel =
+                new ViewModelProvider(this).get(DashboardViewModel.class);
+
+        binding = FragmentDashboardBinding.inflate(inflater, container, false);
+        View root = binding.getRoot();
+        initializeDateCard();
+        createDatePicker(root);
+        initializePieChart(root);
+
+        Duration duration = Duration.between(instant, Instant.now());
+        long millis = duration.toMillis();
+        if (1000 < millis) {
+            Log.w("FiBo", "Creating dashboard view took " + millis + " ms (more than 1 s!)");
+        } else {
+            Log.v("FiBo", "Creating dashboard view took " + millis + " ms");
+        }
```

Comment:
This is because, as we stated in our [SRS](https://github.com/Cebox82/FiBo/blob/fb14dffd3624af414edfda975bfd507a7563bf0e/docs/RUP_Architecture.md#size-and-performance), we employ metrics to make sure our performance is acceptable. :) In this sense, we need to make sure we solve simple problems in < 1000 ms.